ui/core/proto_utils/equipped_item.ts

```diff
@@ -72,6 +84,22 @@ export class EquippedItem {
 		// Make a defensive copy
 		return this._gems.map(gem => (gem == null ? null : Gem.clone(gem)));
 	}
+	get upgrade(): ItemLevelState | null {
+		return this._upgrade;
+	}
```

Comment:
You should be able to declare a pure `ItemLevelState` as the output type, since you have already declared `_upgrade` as a pure `ItemLevelState` rather than `ItemLevelState | null`.